File: colcon_ws/src/pkg_cpp_examples/src/simple_parameters.cpp
#include <rclcpp/rclcpp.hpp>
#include <string>
#include <vector>
#include <rcl_interfaces/msg/set_parameters_result.hpp>
#include <memory>

using std::placeholders::_1;

class SimpleParameter: public rclcpp::Node
{
public:
    SimpleParameter(): Node("simple_param")
    {
        declare_parameter<int>("age_param", 27);
        declare_parameter<std::string>("name_param", "sanjeev");
        param_callback_handle_= add_on_set_parameters_callback(std::bind(&SimpleParameter::param_change_callback, this, _1));
    }

private:
    OnSetParametersCallbackHandle::SharedPtr param_callback_handle_;

    rcl_interfaces::msg::SetParametersResult param_change_callback (const std::vector<rclcpp::Parameter> &parameters)
    {
        rcl_interfaces::msg::SetParametersResult result;

        for (const auto& param: parameters)
        {
            if(param.get_name() == "age_param" && param.get_type() == rclcpp::ParameterType::PARAMETER_INTEGER)
            {
                RCLCPP_INFO_STREAM(get_logger(), "Param age_param changed! New value is: "<< param.as_int());
                result.successful = true;
            }
            else if(param.get_name() == "name_param" && param.get_type() == rclcpp::ParameterType::PARAMETER_STRING)
            {
                RCLCPP_INFO_STREAM(get_logger(), "Param name_apram changed! New value is: "<< param.as_string());
                result.successful = true;
            }
        }

        return result;
    }

};


int main(int argc, char* argv [])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimpleParameter>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: colcon_ws/src/pkg_python_examples/pkg_python_examples/simple_parameters.py
import rclpy
from rclpy.node import Node
from rcl_interfaces.msg import SetParametersResult
from rclpy.parameter import Parameter


class Param(Node):
    """
    This class implements the example to show how we can use the ROS
    parameters to configure some ROS Node
    """
    def __init__(self):
        # this is class constructor
        super().__init__(node_name="simple_python_parameters")
        self.declare_parameter(name="age_param", value=27)
        self.declare_parameter(name="name_param", value="sanjeev")
        
        self.add_on_set_parameters_callback(callback=self.param_change_callback)

    def param_change_callback(self, params):
        """
        This callback function will execute whenever the
        node parameters get changed.

        It will also return the result: in form of changed value of params.
        """
        result = SetParametersResult()

        for param in params:
            if param.name == "age_param" and param.type_ == Parameter.Type.INTEGER:
                self.get_logger().info(f"Param age_param changed! New value is: {param.value}")
                result.successful = True
            if param.name == "name_param" and param.type_ == Parameter.Type.STRING:
                self.get_logger().info(f"Param name_param changed! New value is: {param.value}")
                result.successful = True
        
        return result

def main():
    rclpy.init()
    simple_param = Param()
    rclpy.spin(simple_param)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

            
